### PR TITLE
fix: support escaping of brackets

### PR DESCRIPTION
**Pull request by domoritz**
Fixes https://github.com/vega/vega-lite/issues/5660

### CODE REVIEW

File: src/util.ts [hunk 1]
```diff
@@ -293,6 +293,10 @@ export function titlecase(s: string) {
   return s.charAt(0).toUpperCase() + s.substr(1);
 }
 
+function escapePatchAccess(string: string) {
+  return replaceAll(replaceAll(replaceAll(string, ']', '\\]'), '[', '\\['), '.', '\\.');
```
Comment (kanitw) [inline src/util.ts]: Minor: This isn't very readable

Comment (domoritz) [reply src/util.ts]: What do you suggest? Make a function to replace multiple characters or just a regex?

Comment (kanitw) [reply src/util.ts]: a function? (or ignore this comment for now). 

I'm more interested to understand what's going on.

Comment (domoritz) [reply src/util.ts]: The regex is not that difficult and we use regexes in other places for these kinds of things.

**Comment by domoritz** [comment]
@kanitw https://github.com/vega/vega-lite/issues/5660 explains the issue.

**Pull request merged**